### ISSUE TITLE
OpenMPI version 3.0 and node oversubscription

### ISSUE THREAD

**Issue by jrper** [FluidityProject/fluidity#182]
It see that OpenMPI version 3.0 has changed the default behaviour regarding oversubscription and mpirun. On this version, a plain call like `mpirun -np 8 python -c "print('Hello')"` will fail with an MPI error on systems with fewer than 8 slots available. This includes several of the Fluidity tests, which can request anything up to 16 processes.

The "right way" to do things is now to specify `mpirun -np 8 -oversubscribe python -c "print('Hello')"`, on specific calls, or to define an environment variable, `OMPI_MCA_rmaps_base_oversubscribe=1 mpirun -np 8 python -c "print('Hello')"` to get back to the openmpi 2.0 behaviour. I'm not sure what the right thing to do for us is, possibly to use the last option inside of test harness itself?

**Comment by Patol75** [comment]
From another GitHub project, they ran into the same kind of error (https://github.com/dealii/dealii/issues/5123) and fixed it using the environment variable method (https://github.com/dealii/dealii/pull/5142).
Another idea could be to have the test cases adaptive, in the sense that they would ask the system for the number of available processors, with nproc --all for example, and adjust the number of requested MPI processes accordingly. But this would imply having to modify quite a few test cases xml files, with probably nprocs not being treated as an attribute any more (whenever nprocs > 1) but rather included into the command line element such as:
```
#!/bin/bash
nprocs=16
nprocsLocal=$(nproc --all)
if (( nprocsLocal >= nprocs )); then
  # All fine
else
  # Oversubscribe
fi
```
Or even better, it could stay an attribute if its value is known at runtime (I have no idea if true or not). All of this would just require a simple enough Python script with lxml.

**Comment by jrper** [comment]
Closed by #199, through added environment variable in the testharness.
